dynacred/src/Data.ts

```diff
@@ -446,37 +462,14 @@ export class Data {
     async registerContact(contact: Contact, balance: Long = Long.fromNumber(0),
                           progress: (text: string, width: number) => void = this.dummyProgress): Promise<any> {
         try {
-            progress("Verifying Registration", 10);
-            if (contact.isRegistered()) {
-                return Promise.reject("cannot register already registered contact");
-            }
-            const pub = contact.seedPublic;
-            Log.lvl2("Registering contact", contact.alias,
-                "with public key:", pub.toHex());
-            Log.lvl2("Registering darc");
-            progress("Creating Darc", 20);
-            const d = Contact.prepareUserDarc(pub.point, contact.alias);
-            const darcInstances = await this.spawnerInstance.spawnDarcs(this.coinInstance,
-                [this.keyIdentitySigner], d);
-
-            progress("Creating Coin", 50);
-            Log.lvl2("Registering coin");
-            const coinInstance = await this.spawnerInstance.spawnCoin(this.coinInstance,
-                [this.keyIdentitySigner], darcInstances[0].darc.getBaseID(), contact.seedPublic.toBuffer());
-            let referral = null;
-            if (this.contact.credentialInstance) {
-                referral = this.contact.credentialInstance.id;
-                Log.lvl2("Adding a referral to the credentials");
-            }
-            Log.lvl2("Registering credential");
-
-            progress("Creating Credential", 80);
-            const credentialInstance = await this.createUserCredentials(pub, darcInstances[0].id, coinInstance.id,
-                referral, contact);
-            await this.coinInstance.transfer(balance, coinInstance.id, [this.keyIdentitySigner]);
-            Log.lvl2("Registered user for darc::coin::credential:", darcInstances[0].id, coinInstance.id,
-                credentialInstance.id);
-            await contact.updateOrConnect();
+            const d = new Data(this.bc, this.storage);
```

Comment:
Why passing `this.storage`? `Data.setValues` doesn't use any field of it.

Reply:
It's for a later `save`. But now changed to actually passing the `IStorage` class, as in a later version I'd like to pass the 'file-name' in the storage-class itself.